whois-api/src/main/java/net/ripe/db/whois/api/httpserver/dos/IpBlockListFilter.java

```diff
@@ -0,0 +1,65 @@
+package net.ripe.db.whois.api.httpserver.dos;
+
+import com.google.common.net.InetAddresses;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.FilterConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import net.ripe.db.whois.common.hazelcast.HazelcastBlockedIps;
+import net.ripe.db.whois.common.ip.IpInterval;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+
+@Component
+public class IpBlockListFilter implements Filter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IpBlockListFilter.class);
+
+    private final HazelcastBlockedIps hazelcastBlockedIps;
+
+    public IpBlockListFilter(final HazelcastBlockedIps hazelcastBlockedIps){
+        this.hazelcastBlockedIps = hazelcastBlockedIps;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain) throws IOException, ServletException {
+        final HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+        final HttpServletResponse httpResponse = (HttpServletResponse) servletResponse;
+
+        if (isBlockedIp(httpRequest.getRemoteAddr())){
+            httpResponse.setStatus(HttpStatus.TOO_MANY_REQUESTS_429);
+            httpResponse.getWriter().write("You have been permanently blocked. Please contact support");
```

Comment:
Can you check error message ? I think we should return similar to ACL or blocking. We do give reason in few words there.

Reply:
the tasks says
"You have been permanently blocked... contact support" 

similar to ACL and temporary block
I can put more detail

Reply:
I added something else in the message but I don't think that we should create an ad-hoc code for this.
At least not in this task

Reply:
Can you check this message directly with Ed? I am fine with whatever he says , update the JIRA if its different than whats specified. Thanks

Reply:
I talked to Ed
- Your host instead of Your account
- Include IP addresses 
- Suspected abuse is good because it echos the language in the AUP

Reply:
Done

Reply:
Is the `httpResponse.getWriter()` the best (i.e. recommended) way to return a response body? Is the outputstream properly closed if this is used?

Reply:
Also why does the response `sendError()` not return a response body for DELETE or PUT responses? The method signature seems to indicate it can handle a status code and message, but the response body is empty. Also the method returns successfully without throwing an exception. What happens to the response?

Reply:
httpResponse.getWriter() works fine and it is an alternative to sendError()

Reply:
I would recommend upgrade jetty to 12 and try with sendError again

Reply:
discussed vie slack. We will tackle this when upgrading to jetty 12
https://jetty.org/docs/jetty/12/programming-guide/migration/11-to-12.html

Reply:
Thanks for investigating Miguel